### PR TITLE
Routing, Register,  Database(Customers)

### PR DESCRIPTION
**Pull request by phong-et**
Add logic routing for each tab page 
Add page register, q-Validator and complete reg function 
Change "add" column to "address" of seeder, migration, model

### CODE REVIEW

File: server/schema/customer/def.js [hunk 1]
```diff
@@ -25,5 +25,13 @@ const Customer = `
     username: String!
     password: String!
   }
+  input RegInput {
+    username: String!
+    password: String!
+    passwordConfirm: String!
+    fullName: String,
+    phone: String
+    address:String
```
Comment (hemvanh) [inline server/schema/customer/def.js]: missing space-->  address:" "String

File: src/components/qValidator.vue [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    msg: {
+      default: 'Input value does not match the validation',
+      type: String,
+    },
+  },
+}
+</script>
+<style>
+.error {
```
Comment (hemvanh) [inline src/components/qValidator.vue]: change to et-validator-error

File: src/layouts/MyLayout.vue [hunk 3]
```diff
@@ -1,28 +1,33 @@
 <template>
   <q-layout view="hhh lpr fFf">
-    <q-layout-footer>
+    <q-layout-footer class="max-width-center">
+      <demo-tabs v-if="$q.theme === 'ios'" />
```
Comment (hemvanh) [inline src/layouts/MyLayout.vue]: ???

File: src/layouts/MyLayout.vue [hunk 4]
```diff
@@ -1,28 +1,33 @@
...
       <q-tabs inverted position="bottom" color="brown-6" align="justify">
-        <q-route-tab to="/" default name="news" slot="title" icon="stars" label="Home" />
-        <q-route-tab to="/orders" name="order" slot="title" icon="free_breakfast" label="Order" />
-        <q-route-tab to="/stores" name="account" slot="title" icon="store" label="Store" />
-        <q-route-tab to="/member" name="member" slot="title" icon="account_box" label="Member" />
+        <q-route-tab to="/" default name="news" slot="title" icon="stars" label="Home" class="sml-label" />
```
Comment (hemvanh) [inline src/layouts/MyLayout.vue]: remove sml-label class

File: src/layouts/MyLayout.vue [hunk 5]
```diff
@@ -1,28 +1,33 @@
...
+        <q-route-tab to="/orders" name="order" slot="title" icon="free_breakfast" label="Order" class="sml-label" />
+        <q-route-tab to="/stores" name="account" slot="title" icon="store" label="Store" class="sml-label" />
+        <q-route-tab to="/customer" name="customer" slot="title" icon="account_box" label="Member" class="sml-label" />
       </q-tabs>
     </q-layout-footer>
-    <q-page-container class="et-home">
+    <q-btn rounded color="green" to="/customer/register" label="Join now" class="btn-reg" :class="getIsHiddenRegBtn?'hidden':''" />
```
Comment (hemvanh) [inline src/layouts/MyLayout.vue]: :class="{hidden: getIsHiddenRegBtn}"

File: src/pages/register.vue [hunk 6]
```diff
@@ -1,7 +1,98 @@
...
+    username: {
+      required,
+      minLength: minLength(3),
+      alphaNum,
+    },
+    password: {
+      required,
+      minLength: minLength(3),
+      noSpace: function(value) {
+        var hasWSpace = value.indexOf(' ') >= 0
```
Comment (hemvanh) [inline src/pages/register.vue]: return !( value.indexOf(' ') >= 0)

**Comment by phong-et** [comment]
I fixed and removed some code after reviewing pull request. I committed to my github repo.

**Pull request merged**